docs/doxygen/events.md

```diff
@@ -0,0 +1,75 @@
+### Execution Engine and Events
+
+The execution engine is an execution context that supports event-driven network
+execution on the CUDA streams, CPU threads, and DPU threads. It is intended to
+interact with execution threads that are asynchronous (offloaded collective
+execution) which can be implemented on GPUs, DPUs, or remote CPUs.  
```

Comment:
"execution threads that are asynchronous" -> asynchronous agents? In the case of GPU streams, it may still be a synchronous thread that hands off work to the GPU scheduler right?

Also, does the agent have to be asynchronous? In UCP, there is worker arm/wait where a synchronous thread may arm an agent for work and wakeup on completion. Should that also be in scope for execution engine and events?

Reply:
It doesn't have be asynchronous threads, but for me it makes sense to use in that scenarios. Hopefully, I clarified this in the WG meeting, if not I will be happy to discuss the motivation and usage.